kernel/src/engine/default/filesystem.rs

```diff
@@ -48,9 +45,19 @@ impl<E: TaskExecutor> FileSystemClient for ObjectStoreFileSystemClient<E> {
         path: &Url,
     ) -> DeltaResult<Box<dyn Iterator<Item = DeltaResult<FileMeta>>>> {
         let url = path.clone();
-        let offset = Path::from(path.path());
-        // TODO properly handle table prefix
-        let prefix = self.table_root.child("_delta_log");
+        let offset = Path::from_url_path(path.path())?;
+        let prefix = if url.path().ends_with('/') {
+            offset.clone()
+        } else {
+            let parts = offset.parts().collect_vec();
+            if parts.is_empty() {
+                return Err(Error::generic(format!(
+                    "Offset path must not be a root directory. Got: '{}'",
+                    url.as_str()
+                )));
+            }
+            Path::from_iter(parts[..parts.len() - 1].iter().cloned())
```

Comment:
I think this is just
```suggestion
            let parts = offset.parts().collect_vec();
            if parts.pop().is_empty() {
                return Err(Error::generic(format!(
                    "Offset path must not be a root directory. Got: '{}'",
                    url.as_str()
                )));
            }
            Path::from_iter(parts)
```